The skipIf condition does not appear to work correctly on with
readline-6.0-4.el6.  Just skip it.
http://bugs.python.org/issue19884
http://bugs.python.org/issue22773

diff -Naurp a/Lib/test/test_readline.py b/Lib/test/test_readline.py
--- a/Lib/test/test_readline.py	2014-12-10 09:59:46.000000000 -0600
+++ b/Lib/test/test_readline.py	2014-12-11 21:45:47.049471521 -0600
@@ -44,9 +44,7 @@ class TestHistoryManipulation (unittest.
 
 class TestReadline(unittest.TestCase):
 
-    @unittest.skipIf(readline._READLINE_VERSION < 0x0600
-                     and "libedit" not in readline.__doc__,
-                     "not supported in this library version")
+    @unittest.skip("not supported in this library version")
     def test_init(self):
         # Issue #19884: Ensure that the ANSI sequence "\033[1034h" is not
         # written into stdout when the readline module is imported and stdout
